frontend/control-center/src/pages/Status/ComponentListItem/ItemInfo.tsx

```diff
@@ -1,27 +1,54 @@
 import React, {useState} from 'react';
+
+import {enableDisableComponent} from '../../../actions';
 import {ReactComponent as CheckmarkIcon} from 'assets/images/icons/checkmarkFilled.svg';
 import {ReactComponent as UncheckedIcon} from 'assets/images/icons/serviceUnhealthy.svg';
 import {ReactComponent as ArrowRight} from 'assets/images/icons/arrowRight.svg';
 import {getChannelAvatar} from '../../../components/ChannelAvatar';
 import {getComponentName} from '../../../services';
 import {getSourceForComponent} from 'model';
-import {Toggle} from 'components';
+import {SettingsModal, Button, Toggle} from 'components';
 import styles from './index.module.scss';
+import {connect, ConnectedProps} from 'react-redux';
+import {useTranslation} from 'react-i18next';
 
 type ComponentInfoProps = {
   healthy: boolean;
   itemName: string;
   isComponent: boolean;
   isExpanded: boolean;
   enabled?: boolean;
+} & ConnectedProps<typeof connector>;
+
+const mapDispatchToProps = {
+  enableDisableComponent,
 };
 
-export const ItemInfo = (props: ComponentInfoProps) => {
-  const {healthy, itemName, isComponent, isExpanded, enabled} = props;
+const connector = connect(null, mapDispatchToProps);
+
+const ItemInfo = (props: ComponentInfoProps) => {
+  const {healthy, itemName, isComponent, isExpanded, enabled, enableDisableComponent} = props;
   const [channelSource] = useState(itemName && getSourceForComponent(itemName));
   const [componentName] = useState(itemName && getComponentName(itemName));
   const [componentEnabled, setComponentEnabled] = useState(enabled);
+  const [enablePopupVisible, setEnablePopupVisible] = useState(false);
   const isVisible = isExpanded || isComponent;
+  const {t} = useTranslation();
+
+  const triggerEnableDisableAction = (enabled: boolean) => {
+    enableDisableComponent({components: [{name: itemName, enabled: enabled}]});
+    setComponentEnabled(enabled);
+    setEnablePopupVisible(false);
+  };
+
+  const enableHandler = (enabled: boolean) => {
```

Comment:
Nitpick: The idiomatic way to name these two functions would be something more like `onEnableComponent` and `onToggleEnableComponent` ("Component" optional).